quickstart/src/main/java/com/microsoft/azure/kusto/quickstart/SampleApp.java

```diff
@@ -268,35 +283,63 @@ public class SampleApp {
+            waitForUser = config.isWaitForUser();
 
-            if (config.isIngestData()) {
-                ingestion(config, kustoClient, ingestClient);
+            if (config.getAuthenticationMode() == AuthenticationModeOptions.USER_PROMPT) {
+                waitForUserToProceed("You will be prompted *twice* for credentials during this script. Please return to the console after authenticating.");
             }
-            if (config.isQueryData()) {
-                postIngestionQuerying(kustoClient, config.getDatabaseName(), config.getTableName(), config.isIngestData());
+            try {
+                IngestClient ingestClient = IngestClientFactory.createClient(Utils.Authentication.generateConnectionString(config.getIngestUri(),
+                        config.getAuthenticationMode()));
+                Client kustoClient = ClientFactory
+                        .createClient(Utils.Authentication.generateConnectionString(config.getKustoUri(), config.getAuthenticationMode()));
+
+                preIngestionQuerying(config, kustoClient);
+
+                if (config.isIngestData()) {
+                    ingestion(config, kustoClient, ingestClient);
+                }
+                if (config.isQueryData()) {
+                    postIngestionQuerying(kustoClient, config.getDatabaseName(), config.getTableName(), config.isIngestData());
+                }
+
+            } catch (URISyntaxException e) {
+                Utils.errorHandler("Couldn't create client. Please validate your URIs in the configuration file.", e);
             }
 
-        } catch (URISyntaxException e) {
-            Utils.errorHandler("Couldn't create client. Please validate your URIs in the configuration file.", e);
         }
-
         System.out.println("\nKusto sample app done");
+    }
 
+    private static void createKustoTracer() {
+        enableDistributedTracing();
+        DistributedTracing.initializeTracer(new OpenTelemetryTracer());
+    }
+
+    private static void enableDistributedTracing() {
+        Resource resource = Resource.getDefault()
+                .merge(Resource.create(Attributes.of(ResourceAttributes.SERVICE_NAME, "logical-service-name")));
+        LoggingSpanExporter spanExporter = new LoggingSpanExporter();
+        SdkTracerProvider sdkTracerProvider = SdkTracerProvider.builder()
+                .addSpanProcessor(BatchSpanProcessor.builder(spanExporter).build())
+                .setResource(resource)
+                .build();
+
+        SdkMeterProvider sdkMeterProvider = SdkMeterProvider.builder()
+                .registerMetricReader(PeriodicMetricReader.builder(new LoggingMetricExporter()).build())
+                .setResource(resource)
+                .build();
+        OpenTelemetrySdk.builder()
+                .setTracerProvider(sdkTracerProvider)
+                .setMeterProvider(sdkMeterProvider)
+                .setPropagators(ContextPropagators.create(W3CTraceContextPropagator.getInstance()))
```

Comment:
What is w3cTracecontext propagator?

Do we really need all of these for this to work
Make sure it's the most minimal thing possible